### PR TITLE
Drop old Python and Django versions

### LINKED ISSUE #176: Drop `Python <= 3.7` and `Django <= 2.1` support.

### PR DESCRIPTION
**Pull request by merwok**
fixes #176

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,15 +52,12 @@ django-admin-interface is a modern **responsive flat admin interface customizabl
 INSTALLED_APPS = (
     #...
     "admin_interface",
-    "flat_responsive", # only if django version < 2.0
-    "flat", # only if django version < 1.9
```
Comment (merwok) [inline README.md:56]: Should add a migration note telling people to remove these?

File: tests/test_settings.py [hunk 2]
```diff
@@ -27,22 +22,15 @@ def tearDown(self):
         pass
 
     def __test_installed_apps(self):
```
Comment (merwok) [inline tests/test_settings.py:24]: I would understand the tests in this file better if the style was closer to pytest markers, where you see one function for each case with parameters showing clearly the django version, the installed apps and the expected error 🙂 

Could you add a commit to this branch with the correct fix?

Comment (fabiocaccamo) [reply tests/test_settings.py:24]: I don't understand why, but I'm not able to push on this branch, anyway, here the `test_settings` file updated:

```python
from django.conf import settings
from django.core.exceptions import ImproperlyConfigured
from django.test import TestCase, override_settings

from admin_interface.settings import check_installed_apps


class AdminInterfaceSettingsTestCase(TestCase):

    DJANGO_APPS = [
        "django.contrib.admin",
        "django.contrib.auth",
        "django.contrib.contenttypes",
        "django.contrib.messages",
        "django.contrib.sessions",
    ]

    def setUp(self):
        pass

    def tearDown(self):
        pass

    def __test_installed_apps(self):
        installed_apps = settings.INSTALLED_APPS

        if "colorfield" not in installed_apps:
            self.assertRaises(ImproperlyConfigured, check_installed_apps)
        else:
            check_installed_apps()

    @override_settings(
        INSTALLED_APPS=[
            "admin_interface",
            "colorfield",
        ]
        + DJANGO_APPS
    )
    def test_installed_apps_all(self):
        self.__test_installed_apps()

    @override_settings(
        INSTALLED_APPS=[
            "admin_interface",
            # 'colorfield',
        ]
        + DJANGO_APPS
    )
    def test_installed_apps_no_colorfield(self):
        self.__test_installed_apps()
```

File: README.md (hunk 1 above)
Comment (fabiocaccamo) [reply README.md:56]: mhmh no... even if they are installed nothing happens.

File: admin_interface/compat.py [hunk 3]
```diff
@@ -1,26 +1,12 @@
...
+from django.utils.encoding import force_str
+from django.utils.translation import gettext_lazy
+
+__all__ = [
+    "FileExtensionValidator",
+    "NoReverseMatch",
+    "reverse",
+    "force_str",
+    "gettext_lazy",
+]
```
Comment (fabiocaccamo) [inline admin_interface/compat.py:12]: This module could be removed at all now, just need to replace imports where this module is used.

**Review by fabiocaccamo — DISMISSED** [review]

**Review by fabiocaccamo — APPROVED** [review]

**Pull request merged**